src/ogc/wcs/WcsCoverageDescriptions.js

```diff
@@ -59,15 +61,126 @@ define([
             this.assembleDocument();
         };
 
+        /**
```

Comment:
Did you consider defining a `WcsCoverageDescription` object model that could encapsulate the responsibility of `getSector`, `getResolution`, and `getSupportedCrs`?

Reply:
No I hadn't thought about that approach. With the current approach, we've eliminated the need for a user to know the difference between versions and the index location. If those functions were refactored to singular WcsCoverageDescription(s) the notion of version or index would be required by the client unless the functionality was duplicated in parent WcsCoverageDescriptions.

What considerations am I missing in my thinking?

Reply:
It seems to me there is some conceptual fuzziness in the WCS object model. In particular the relationship of WcsCoverageDescriptions and WcsCoverage. Here are two points I noticed that motivated me to comment:

• Extending WcsCoverage to facilitate future convenience methods requires changes to WcsCoverageDescriptions. That makes WcsCoverage look partially like a leaky abstraction, since for the purpose of extracting XML doc data WcsCoverage and WcsCoverageDescriptions are tightly coupled.
• Apps cannot easily query the WcsCoverage for information in the XML doc.

I can see why we'd want a WcsCoverage object model that's logically independent from the coverage summaries and coverage descriptions coming from the XML documents. The coverage descriptions for WCS 1.0 and 2.0 are quite different, but what if the common concept was represented by an object model in a manner similar to WcsCoverageDescriptions?